projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -688,8 +693,8 @@ private[projection] class R2dbcOffsetStore(
 
   def deleteOldTimestampOffsets(): Future[Int] = {
     val currentState = getState()
-    if (currentState.window.compareTo(settings.timeWindow) < 0) {
-      // it hasn't filled up the window yet
+    if (idle.getAndSet(true) || currentState.window.compareTo(settings.timeWindow) < 0) {
+      // no new offsets stored since previous delete, or it hasn't filled up the window yet
```

Comment:
Hmm, I'm not sure if the logic here is correct. 

If idle is `false`, the first if branch won't pass, but we side-effect the atomic boolean. If the second branch pass the test, we won't be running any delete and now we put idle back to `true`.

Reply:
Well, I guess in that case we lose one chance to run the deletion, but it will come back later anyway.

Reply:
thanks, I changed it: https://github.com/akka/akka-persistence-r2dbc/pull/179/commits/b934ee387a970adba0729dffe5f443e8185155da